### ISSUE TITLE
Data preprocessing

### ISSUE THREAD

**Issue by Miraclerice** [xmed-lab/FSDiffReg#6]
Hi, if it's convenient for you, can you provide the preprocessing code of your dataset? @Eason-Qin

**Comment by Eason-Qin** [comment]
Thank you for your interest in our work! I can provide the details of the preprocessing for you, and it should be fast to implement one:

To improve training efficiency, the training dataset is preprocessed by cropping and resampling before putting it to training pipeline. The preprocessing steps are consistent with DiffuseMorph. Specifically, for resampling, the SimpleITK BSpline Resampler is used to resample the image to (1.5, 1.5, 3.15) Voxel Spacing. For cropping, SimpleITK is used to center crop to (128, 128), and the processing of the z-axis is already written in the dataloader. The order is to resample first and then crop. The validation is performed using patients 090-100, and the same training/validation split is used for all baselines and our method.

Reference:
Kim, B., Han, I. and Ye, J.C., 2022, October. DiffuseMorph: unsupervised deformable image registration using diffusion model. In European Conference on Computer Vision (pp. 347-364). Cham: Springer Nature Switzerland.

**Comment by lxy51** [comment]
Hello, is there all the specific code for preprocessing

**Comment by cyl0000** [comment]
在重采样到 (1.5, 1.5, 3.15)后有些数据的z坐标并不足以到32，这种情况要如何处理呢
